csaf_2.1/prose/edit/src/tests-01-mndtr-11-cwe.md

```diff
@@ -1,20 +1,24 @@
 ### CWE
 
-It MUST be tested that given CWE exists and is valid.
+It MUST be tested that given CWE exists and is valid in the version provided.
+Any `id` that refers to a CWE Category or View MUST fail the test.
 
 The relevant path for this test is:
 
 ```
-    /vulnerabilities[]/cwe
+    /vulnerabilities[]/cwes[]
 ```
 
 *Example 1 (which fails the test):*
 
 ```
-  "cwe": {
-    "id": "CWE-79",
-    "name": "Improper Input Validation"
-  }
+  "cwes": [
```

Comment:
We could ease access per using an object as collector (assuming advisories will not refer to one CWE ID with different versions):
```json
  "cwe": {
    "CWE-79": {
      "name": "Improper Input Validation",
      "version": "4.13"
    }
  }
```

Reply:
> assuming advisories will not refer to one CWE ID with different versions

I'm not 100% sure about that yet :see_no_evil:

Reply:
This could potentially be realized with pattern properties:

```
       "cwes": {
            "title": "List of CWEs",
            "description": "Contains a list of CWEs.",
            "type": "object",
            "patternProperties": {
               "^CWE-[1-9]\\d{0,5}$": { 
                          "title": "CWE",
                          "description": "Holds the MITRE standard Common Weakness Enumeration (CWE) for the weakness associated.",
                          "type": "object",
                          "required": [
                             "name",
                             "version"
                          ],
                         "properties": {
                             "name": {
                                // ...
                             },
                             "version": {
                                // ...
                             }
                         }
              }
         }
```

I'm not sure whether that is really an access simplification: Usually, it is hard for rendering engines (that convert JSON to rendered HTML based on a template) to access JSON keys that they don't know the name of resp. iterate over properties. However, iterating over an array is a standard feature.